src/map.tsx

```diff
@@ -203,13 +203,9 @@ export default class ReactMapboxGl extends React.Component<Props & Events, State
     const { map } = this.state as State;
 
     if (map) {
-      // Remove all events attached to the map
-      map.off();
-
-      // NOTE: We need to defer removing the map to after all children have unmounted
-      setTimeout(() => {
-        map.remove();
-      });
+      // https://www.mapbox.com/mapbox-gl-js/api/#map#remove
```

Comment:
Yep. From [the link](https://www.mapbox.com/mapbox-gl-js/api/#map#remove) I put in the comments:

> Clean up and release all internal resources associated with this map.
> This includes DOM elements, event bindings, web workers, and WebGL resources.